src/menus/serverCreationScreen.cpp

```diff
@@ -77,7 +78,10 @@ ServerSetupScreen::ServerSetupScreen()
     // Close server button.
     (new GuiButton(this, "CLOSE_SERVER", tr("Close"), [this]() {
         destroy();
-        returnToMainMenu();
+        if (PreferencesManager::get("server_scenario")=="")
```

Comment:
I'm not liking this hack. I think this feature is easier implemented here:
https://github.com/daid/EmptyEpsilon/blob/master/src/main.cpp#L331
Then you won't need this hack, and only open the scenario once on startup.

Reply:
Okay, will change it. Thought it will be nice to have all those startup-related options at one place, but requiring that hack then kinda defeats the purpose. Doing that before the initial main menu call would also prioritise that feature when multiple options are set, but I guess that won't matter much.